lxd/instance_post.go

```diff
@@ -454,11 +454,33 @@ func instancePostClusteringMigrate(d *Daemon, inst instance.Instance, oldName, n
 			Mode: "pull",
 		}
 
-		copyOp, err := dest.CopyInstance(source, *entry, &args)
+		copyOp, migOpId, err := dest.CopyInstance(source, *entry, &args)
 		if err != nil {
 			return errors.Wrap(err, "Failed to issue copy instance API request")
 		}
 
+		handler := func(event api.Event) {
```

Comment:
I suspect this part could be done by using `copyOp.AddHandler` which will then be called every time the operation changes, that handler would then extract the Metadata part of that operation and in turn pass it to op.UpdateMetadata.

Reply:
@stgraber Maybe I'm not understand something but `copyOp.AddHandler` won't work because `copyOp` is operation of type 'Create Instance' and it doesn't generate progress events. Progress events are generated by 'Migrate instance' operation which is created here: [`source.MigrateInstance`](https://github.com/lxc/lxd/blob/f12f03a4ba4645892ef6cc167c24da49d1217b02/client/lxd_instances.go#L521) (this is why I tried to pass ID of this operation from `CopyInstance`). I understand that we can't change functions in `client/interfaces.go` so I'm searching other solution. If you have any ideas it will be helpfull.

Reply:
@stgraber `copyOp.AddHandler` started working when i changed [`TrackProgress`](https://github.com/lxc/lxd/blob/master/lxd/migrate_instance.go#L878) from `false` to `true` but is it a good direction?

Reply:
Yes, I think that's correct. It's how it's done for the storage volumes and given that we rely on the transfer status of the receiver and not the sender, having TrackProgress set to false on the receiver feels wrong.

Can you make a separate commit in your branch which flips it from `false` to `true`, remove the `Do not use ...` comment next to it and indicate in the commit message that for migration we need the receiving side to send progress information?

Then the rest of your branch should be pretty easy from that point on as you'll be getting the right events :)